### ISSUE TITLE
 (!copied) is equivalent to (hdr->b_l1hdr.b_freeze_cksum == NULL)

### ISSUE THREAD

**Issue by nedbass** [openzfs/zfs#6527]
### System information

Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | TOSS (RHEL 7.3 derivative)
Distribution Version    | toss-release-3.1-4.1rc2
Linux Kernel                 | 3.10.0-514.26.2.1chaos.ch6_1.x86_64
Architecture                 | x86_64
ZFS Version                  | v0.7.0-31_gc8f9061
SPL Version                  | v0.7.0-12_g9df9692

### Describe the problem you're observing
System panics when generating a send stream.

### Describe how to reproduce the problem

I hit the problem running the following script from #6524 under a debug build. The command that panics is `$ZFS send $DS@a > senda`

```
#!/bin/bash

ZFS=./cmd/zfs/zfs
ZPOOL=./cmd/zpool/zpool
POOL=tank
DS=$POOL/fish
ROOTMTPT=/mnt/$POOL
MTPT=/mnt/$DS
NEWDS=$POOL/new
RFILE=./rfile

dd if=/dev/urandom of=$RFILE bs=262144 count=1

echo 1 > /sys/module/zfs/parameters/zfs_dbgmsg_enable
echo 512 > /sys/module/zfs/parameters/zfs_flags

truncate -s 10g /tmp/a
$ZPOOL create -f -O mountpoint=$ROOTMTPT $POOL /tmp/a

mkfiles () {
        for ((i=0;i<5000;i++)) ; do
                dd if=$RFILE of=$MTPT/file-$i bs=262144 > /dev/null
        done
}

$ZFS create -o dnsize=auto $DS
mkfiles
umount $MTPT
$ZFS snap $DS@a

$ZFS set dnsize=legacy $DS
$ZFS mount $DS
rm $MTPT/*
mkfiles
umount $MTPT
$ZFS snap $DS@b

$ZFS send $DS@a > senda
$ZFS send -i $DS@a $DS@b > sendb
$ZFS recv $NEWDS < senda
$ZFS recv $NEWDS < sendb
```

### Include any warning/errors/backtraces from the system logs

```

[  367.132385] SPL: Loaded module v0.7.0-12_g9df9692
[  367.178933] SPLAT: Loaded module v0.7.0-12_g9df9692
[  369.264850] ZFS: Loaded module v0.7.0-31_gc8f9061 (DEBUG mode), ZFS pool version 5000, ZFS filesystem version 5
[  404.202115] (!copied) is equivalent to (hdr->b_l1hdr.b_freeze_cksum == NULL)
[  404.209827] PANIC at arc.c:1811:arc_buf_try_copy_decompressed_data()
[  404.218359] Showing stack for process 173266
[  404.223695] CPU: 13 PID: 173266 Comm: lt-zfs Tainted: P           OE  ------------   3.10.0-514.26.2.1chaos.ch6_1.x86_64 #1
[  404.236714] Hardware name: Intel Corporation S2600WTTR/S2600WTTR, BIOS SE5C610.86B.01.01.0016.033120161139 03/31/2016
[  404.236717]  ffffffffa0fc58bc 00000000161d01d4 ffff881f2f76b998 ffffffff8169d4bc
[  404.236719]  ffff881f2f76b9a8 ffffffffa02c32d4 ffff881f2f76bb30 ffffffffa02c33a9
[  404.236720]  ffff881fedb03b80 ffff880f00000020 ffff881f2f76bb40 ffff881f2f76bae0
[  404.236721] Call Trace:
[  404.236738]  [<ffffffff8169d4bc>] dump_stack+0x19/0x1b
[  404.236748]  [<ffffffffa02c32d4>] spl_dumpstack+0x44/0x50 [spl]
[  404.236751]  [<ffffffffa02c33a9>] spl_panic+0xc9/0x110 [spl]
[  404.236756]  [<ffffffff816b0171>] ? ftrace_call+0x5/0x2f
[  404.236757]  [<ffffffff816b0171>] ? ftrace_call+0x5/0x2f
[  404.236760]  [<ffffffffa02bf359>] ? spl_kmem_cache_alloc+0x99/0x150 [spl]
[  404.236808]  [<ffffffffa0e09d75>] arc_buf_fill+0x575/0x1750 [zfs]
[  404.236810]  [<ffffffff816b0171>] ? ftrace_call+0x5/0x2f
[  404.236814]  [<ffffffffa02bf359>] ? spl_kmem_cache_alloc+0x99/0x150 [spl]
[  404.236817]  [<ffffffffa02bf359>] ? spl_kmem_cache_alloc+0x99/0x150 [spl]
[  404.236828]  [<ffffffffa0e0b283>] arc_buf_alloc_impl+0x313/0x6e0 [zfs]
[  404.236838]  [<ffffffffa0e0d15b>] arc_read+0xefb/0x1990 [zfs]
[  404.236840]  [<ffffffff816b0171>] ? ftrace_call+0x5/0x2f
[  404.236851]  [<ffffffffa0e149b0>] ? arc_buf_destroy+0x330/0x330 [zfs]
[  404.236867]  [<ffffffffa0e39acf>] dmu_objset_open_impl+0x1bf/0xb30 [zfs]
[  404.236881]  [<ffffffffa0e3a594>] dmu_objset_from_ds+0x154/0x290 [zfs]
[  404.236909]  [<ffffffffa0eedc3c>] zfs_ioc_snapshot_list_next+0x17c/0x200 [zfs]
[  404.236930]  [<ffffffffa0ef7425>] zfsdev_ioctl+0x655/0x790 [zfs]
[  404.236961]  [<ffffffffa0ef6dd0>] ? pool_status_check+0x140/0x140 [zfs]
[  404.236966]  [<ffffffff8121bba5>] do_vfs_ioctl+0x305/0x510
[  404.236969]  [<ffffffff8121be51>] SyS_ioctl+0xa1/0xc0
[  404.236970]  [<ffffffff816ae449>] system_call_fastpath+0x16/0x1b
```
```
[root@jet22:~]# cat /proc/`pidof lt-zfs`/stack
[<ffffffffa02c33d5>] spl_panic+0xf5/0x110 [spl]
[<ffffffffa0e09d75>] arc_buf_fill+0x575/0x1750 [zfs]
[<ffffffffa0e0b283>] arc_buf_alloc_impl+0x313/0x6e0 [zfs]
[<ffffffffa0e0d15b>] arc_read+0xefb/0x1990 [zfs]
[<ffffffffa0e39acf>] dmu_objset_open_impl+0x1bf/0xb30 [zfs]
[<ffffffffa0e3a594>] dmu_objset_from_ds+0x154/0x290 [zfs]
[<ffffffffa0eedc3c>] zfs_ioc_snapshot_list_next+0x17c/0x200 [zfs]
[<ffffffffa0ef7425>] zfsdev_ioctl+0x655/0x790 [zfs]
[<ffffffff8121bba5>] do_vfs_ioctl+0x305/0x510
[<ffffffff8121be51>] SyS_ioctl+0xa1/0xc0
[<ffffffff816ae449>] system_call_fastpath+0x16/0x1b
[<ffffffffffffffff>] 0xffffffffffffffff
```

**Comment by nedbass** [comment]
Duplicate of #5721

**Comment by behlendorf** [comment]
Closing, resolved by #8979.
